bezier-rs/lib/src/lib.rs

```diff
@@ -853,6 +838,159 @@ impl Bezier {
 		result
 	}
 
+	/// Approximate a bezier curve with circular arcs.
+	/// The algorithm can be customized using the [ArcsOptions] structure by setting the following fields within ArcOptions:
+	/// - maximize_arcs - [MaximizeArcs] - The default value is `Automatic`.,
+	/// - error: f64 - The error used for approximating the arc's fit. The default is `0.5`.
+	/// - max_iterations: i32 - The maximum number of segment iterations used as attempts for arc approximations. The default is `100`.
+	pub fn arcs(&self, arcs_options: ArcsOptions) -> Vec<CircleArc> {
+		let ArcsOptions { maximize_arcs, error, max_iterations } = arcs_options;
+		match maximize_arcs {
+			MaximizeArcs::Automatic => {
+				let (auto_arcs, final_low_t) = self.arcs_helper(0., 1., error, max_iterations, true);
+				if final_low_t != 1. {
+					[
+						auto_arcs,
+						self.split(final_low_t)[1].arcs(ArcsOptions {
+							maximize_arcs: MaximizeArcs::Off,
+							error,
+							max_iterations,
+						}),
```

Comment:
Assign this to a variable, please, so the `[auto_arcs, your_new_variable].concat()` can be on a single line.